### PR TITLE
docs(issue-template): draft 'definition of done', update 'CONTRIBUTING'

### PR DESCRIPTION
**Pull request by SimonFinney**
Contributes to #272 

#### Changelog

**New**

Add 'definition of done' to component or pattern issue template

**Changed**

Add additional clarity to component or pattern issue template and `CONTRIBUTING.md`

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/component-or-pattern.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+- [ ] Delivery follows the guidance in
+      [Contributing Guide](https://github.com/carbon-design-system/ibm-cloud-cognitive/blob/master/.github/CONTRIBUTING.md#5-make-a-pull-request)
+- [ ] Feedback from collaborators is addressed, and any follow-up issues are
+      created
+
+#### Testing
+
+- [ ] Tests are behavior-driven and based on interactions defined in the design
+- [ ] Inputs and outputs are tested
+- [ ] Accessibility tests for each use case
```
Comment (matthewgallo) [inline .github/ISSUE_TEMPLATE/component-or-pattern.md:42]: Might be helpful if we could provide an example of how to run this accessibility test. Or if this test is to be written for each component, an example of one in use.

File: .github/ISSUE_TEMPLATE/component-or-pattern.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      [customization controls](https://storybook.js.org/docs/react/essentials/controls)
+      for interacting with inputs and modifiers
+- [ ] Additional usage documentation is accessible via the 'Docs' tab where
+      applicable
+
+#### Consistency
+
+- [ ] Props are consistent with Carbon-defined conventions, or otherwise
+      internally where unavailable
+- [ ] Custom class names can be supplied for additional style customization
```
Comment (lee-chase) [inline .github/ISSUE_TEMPLATE/component-or-pattern.md:65]: What are props not used to customize as per Carbon?

File: .github/ISSUE_TEMPLATE/component-or-pattern.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+- [ ] Additional props are
```
Comment (lee-chase) [inline .github/ISSUE_TEMPLATE/component-or-pattern.md:66]: All props that have a functional/visual impact should be passed explicitly. What type of additional prop are we referring too?

**Review by lee-chase — APPROVED** [review]

**Review by lee-chase — CHANGES_REQUESTED** [review]
See comments

**Pull request closed without merging**